cmd/gdeploy/commands/provider/provider.go

```diff
@@ -10,5 +14,35 @@ var Command = cli.Command{
 	Subcommands: []*cli.Command{
 		&addCommand, &removeCommand, &updateCommand,
 	},
-	Action: cli.ShowSubcommandHelp,
+	Action: cli.ActionFunc(func(ctx *cli.Context) error {
+		cli.ShowSubcommandHelp(ctx)
+		DeprecatedWarn(ctx)
```

Comment:
Only show this on the add command, and only when no "with" flag arguments are provided
```suggestion
```